src/components/rbac/permission-categories.tsx

```diff
@@ -0,0 +1,143 @@
+      groups: [],
+    };
+  }
+
+  componentDidMount() {
+    const { model_permissions } = this.context.user;
+    this.setState({ groups: this.formatPermissions(model_permissions) });
+  }
+
+  render() {
+    const { groups } = this.state;
+    const { role, showEmpty, showCustom } = this.props;
+
+    const origGroups = groups.map((group) => ({
+      ...group,
+      label: i18n._(group.label),
+    }));
+
+    const allGroups = showCustom
+      ? [
+          ...origGroups,
+          {
+            label: t`Custom permissions`,
+            object_permissions: this.customPermissions(role),
+          },
+        ]
+      : origGroups;
+
+    const groupsToMap = showEmpty
+      ? allGroups
+      : allGroups.filter((group) => this.getSelected(group).length);
+
+    return (
+      <React.Fragment>
+        {groupsToMap.map((group) => (
+          <Flex
+            style={{ marginTop: '16px' }}
+            alignItems={{ default: 'alignItemsCenter' }}
+            key={group.label}
+            className={group.label}
+          >
+            <FlexItem style={{ minWidth: '200px' }}>
+              {i18n._(group.label)}
+            </FlexItem>
+            <FlexItem grow={{ default: 'grow' }}>
+              <PermissionChipSelector
+                availablePermissions={group.object_permissions
+                  .filter(
+                    (perm) =>
+                      !role.permissions.find((selected) => selected === perm),
+                  )
+                  .map((permission) => this.getNicenames(permission))
+                  .sort()}
+                selectedPermissions={role.permissions
+                  .filter((selected) =>
+                    group.object_permissions.find((perm) => selected === perm),
+                  )
+                  .map((permission) => this.getNicenames(permission))}
+                menuAppendTo='inline'
+                multilingual={true}
+                isViewOnly={true}
+              />
+            </FlexItem>
+          </Flex>
+        ))}
+      </React.Fragment>
+    );
+  }
+
+  private formatPermissions(permissions): PermissionType[] {
+    const formattedPermissions = {};
+    for (const [key, value] of Object.entries(permissions)) {
+      if (value['ui_category'] in formattedPermissions) {
+        formattedPermissions[value['ui_category']]['object_permissions'].push(
+          key,
+        );
+      } else {
+        formattedPermissions[value['ui_category']] = {
+          label: value['ui_category'],
+          object_permissions: [key],
+        };
+      }
+    }
+    const arrayPermissions = Object.values(
+      formattedPermissions,
+    ) as PermissionType[];
+    return arrayPermissions;
+  }
+
+  private getNicenames(permission) {
+    const { model_permissions } = this.context.user;
+    if (model_permissions[permission].name !== undefined) {
+      return model_permissions[permission].name;
+    } else {
+      return undefined;
+    }
+  }
```

Comment:
this doesn't work with `showCustom={true}` .. you can try by switching to true in RoleList and expanding the `galaxy.synclist_owner` role ... it will call `getNicenames('galaxy.add_synclist')` and fail because it's missing from `model_permissions`

I guess the quick fix is

```suggestion
  private getNicenames(permission) {
    const { model_permissions } = this.context.user;
    return model_permissions[permission]?.name || permission;
  }
```

, or we could call nicenames only on known permissions instead, up to you :)